cs/datatypes/ref.adoc

```diff
@@ -0,0 +1,222 @@
+= Datový typ `ref!`
+:toc:
+:numbered: 
+
+== Úvodem
+
+Hodnotami typu `ref!` jsou pojmenované odkazy, jako jsou názvy účtů u sociálních medií nebo identifikátory s implikovaným názvem domény.
+
+Datový typ `ref!` patři do typesetu `any-string!`, přičemž je konceptuálně blízký datovému typu `issue!`, od něhož se liší v implementaci (string-like vs. symbol-like) a v poskytované sadě kompromisů.
+
+== Vytvoření
+
+Hodnoty typu `ref!` lze vytvořit s použitím literálové syntaxe při načítání skriptu nebo při runtime konstruktorem `make` či konverzí `to`.
+
+=== Literálová skladba
+
+*Syntaxe*
+
+----
+<ref> format : @<name>
+
+<name> : volitelný název odkazu
+----
+
+* `<name>` nula či více znaků jiných než: `# $ ' , = > @ \ ^`;
+* literálová forma je ukončena jedním z těchto způsobů: `" ( ) ; < [ ] { }` nebo mezerou (space, tab, `CR`, `LF`) .
+
+*Příklady*
+
+----
+@
+@red_lang
+@dockimbel
+@§2.1
+----
+
+=== Konstruktor `make`
+
+Při použití konstruktoru `make` nutno respektovat tato pravidla:
+
+* specifikace `float!` a `integer!` označují počet před-alokovaných bajtů (`float!` is truncated);
+* konstrukce z každé jiné specifikace je ekvivalentní <<Konverze `to`, konverzi>> `to`.
+
+*Příklady*
+
+----
+make ref! 9.2
+== @
+
+make ref! 14
+== @
+
+make ref! <ref>
+== @ref
+
+make ref! [red (_lang)]
+== @red_lang
+
+make ref! #{76616C7565}
+== @value
+
+make ref! none
+** cannot make error
+
+make ref! #datatype
+== @datatype
+----
+
+=== Konverze `to`
+
+Datový typ `ref!` podporuje řadu typových konverzí, platných pro typeset `any-string!`. Při konverzi dané hodnoty na `ref!` jsou použity níže popsané transformace - s předsazením znaku `@`.
+
+.Datové typy, konvertovatelné na `ref!`
+[options="header" cols="1,5"]
+|===
+| Datatype | Description
```

Comment:
"Datatype", "description".

Reply:
It's partly  a  didactical intention:  force a rader to learn english words.
Can be changed.